lib/openscap_parser/value.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+module OpenscapParser
+  class Value < XmlNode
+    include OpenscapParser::Util
+
+    def id
+      @id ||= parsed_xml['id']
+    end
+
+    def description
+      @description ||= newline_to_whitespace(parsed_xml.at_css('description')&.text)
+    end
+
+    def title
+      @title ||= parsed_xml.at_css('title')&.text
+    end
+
+    def type
+      @type ||= parsed_xml['type'] || 'string'
+    end
+
+    def lower_bound(selector = nil)
+      @lower_bounds ||= parsed_xml.xpath('lower-bound').each_with_object({}) do |element, lb|
+        lb[element.at_xpath('@selector')&.text] = element&.text
+      end
+      selector ? @lower_bounds[selector] : @lower_bounds[nil] || @lower_bounds[""]
+    end
+
+    def upper_bound(selector = nil)
+      @upper_bounds ||= parsed_xml.xpath('upper-bound').each_with_object({}) do |element, ub|
+        ub[element.at_xpath('@selector')&.text] = element&.text
+      end
+      selector ? @upper_bounds[selector] : @upper_bounds[nil] || @upper_bounds[""]
+    end
+
+    def value(selector = nil)
+      @values ||= parsed_xml.xpath("value").each_with_object({}) do |element, rv|
+        rv[element.at_xpath('@selector')&.text] = element&.text
+      end
```

Comment:
I tried `parsed_xml.xpath('value').index_by(&:selector)` but `index_by` wouldn't work on the `Nokogiri::XML::NodeSet`

Reply:
https://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri/XML/NodeSet#to_a-instance_method